File: backend/shared/lib/src/main/java/br/ufpr/dac/shared/dto/ContasDto.java
package br.ufpr.dac.shared.dto;

import java.time.LocalDate;

import br.ufpr.dac.shared.dto.GerentesDto.Gerente;
import br.ufpr.dac.shared.dto.UsersDto.Cliente;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

public class ContasDto {

  @Data
  @Builder
  @AllArgsConstructor
  @NoArgsConstructor
  public static class Conta {
    private Long id;
    private String numero;
    private Cliente cliente;
    private Gerente gerente;
    private Double saldo;
    private Double limite;
    private LocalDate dataCriacao;

  }

}

File: backend/apiGatewayServiceNode/src/dto/ContasDto.ts
import { GerentesDtoGerente } from "./GerentesDto";
import { UsersDtoCliente } from "./UsersDto";

export interface ContasDtoConta {
  id: null | number;
  numero: null | string;
  cliente: null | UsersDtoCliente;
  gerente: null | GerentesDtoGerente;
  saldo: null | number;
  limite: null | number;
  dataCriacao: null | string;
}
